### ISSUE TITLE
detach open pdf files from emacs

### ISSUE THREAD

**Issue by dlebrecht** [tmalsburg/helm-bibtex#93]
the current behavior i get: the pdf files opened from emacs, are closed if quiting emacs. 

is it possible to change that behavior, that is, open the files externally and independent of whether emacs is running or not

i suspect my emacs gets slow after having some 10 pdf open. i might be wrong though

**Comment by jagrg** [comment]
Have you tried configuring `helm-bibtex-pdf-open-function `? There's a section on the manual explaining how to do it.

**Comment by dlebrecht** [comment]
i didn't express myself correctly, i suppose.

on my .emacs, i have:

  (setq helm-bibtex-pdf-open-function
        (lambda (fpath)
          (start-process "zathura" "*helm-bibtex-zathura*" "xdg-open"
fpath)))

the files open on the external pdf viewer, this is not the issue. The issue
is if i close emacs, those files are processes in emacs, and they close as
well. Can i change this behavior?

best
d

**Comment by tmalsburg** [comment]
Sounds like a more general Emacs question.  Perhaps ask on https://emacs.stackexchange.com/  If you find a solution, it would be great if you could report it.  I will then update the documentation.  Thank you.

**Comment by tmalsburg** [comment]
Perhaps you can use `start-process-shell-command` and append an `&` to the command.

**Comment by dlebrecht** [comment]
the solution is to use call-process

  (setq helm-bibtex-pdf-open-function
        (lambda (fpath)
          (call-process "xdg-open" nil 0 nil fpath)))



reference: https://www.emacswiki.org/emacs/ExecuteExternalCommand
Executing external programs as their own process

   - Using shell-command runs the app as a child, even when done
   asynchronously
   - start-process also runs as a child process

Both commands result in the child process being killed when emacs is
shutdown. For a command to have the ability to stay alive past that point
you’ll need to use call-process to start the program as its own distinct
process.

(call-process PROGRAM &optional INFILE BUFFER DISPLAY &rest ARGS)

Setting BUFFER to 0 runs the program asynchronously, while nil will wait
for the process to exits before resuming emacs.

**Comment by tmalsburg** [comment]
Great find, thank you.  Emacs's own documentation for `call-process` is unfortunately terribly misleading about this.

**Comment by tmalsburg** [comment]
Documentation changed in 7f6c98a8afd4b3a8e5a3a9850071487edd66b9e1

**Comment by jagrg** [comment]
@tmalsburg you need to separate the `-a` argument, otherwise it won't work on a mac:

```
(setq helm-bibtex-pdf-open-function
      (lambda (fpath)
	(call-process "open" nil 0 nil "-a" "Skim.app" fpath)))
```
If you don't mind me asking, what are the benefits of using `call-process` exactly?

**Comment by tmalsburg** [comment]
With `call-process` PDF viewers can outlive the Emacs process.  On OSX, the problem is perhaps already solved by the open command but if you user other means to open PDFs, quitting Emacs will quit the PDF viewer.

**Comment by tmalsburg** [comment]
`-a` problem fixed in 711604177b2d098071740fb6df94c4872fbc6057.  Thanks.
